### PR TITLE
Add featured articles functionality.

### PR DESCRIPTION
**Pull request by akatsoulas**
* Returns 4 random articles out of the 10 most visited.
* Return 4 random articles out of the 10 most visited per product.
* Blacken some files.

### CODE REVIEW

File: kitsune/landings/jinja2/landings/home.html [hunk 1]
```diff
@@ -1,11 +1,6 @@
-{% extends "base.html" %}
-{% from 'products/includes/product_macros.html' import product_cards with context %}
-{% set styles = ('home',) %}
-{% set canonical_url = canonicalize(viewname='home') %}
-
-{% block hidden_search_masthead %}{% endblock %}
-
-{% block contentwrap %}
+{% extends "base.html" %} {% from 'products/includes/product_macros.html' import product_cards with
```
Comment (johngian) [inline kitsune/landings/jinja2/landings/home.html]: This multiline doesn't look very nice

File: kitsune/landings/jinja2/landings/home.html [hunk 2]
```diff
@@ -41,10 +36,4 @@ <h2 class="text-display-xs">{{ _('Featured Articles & Threads') }}</h2>
...
   </div>
 </section>
-{% endblock %}
-
-{% block outer_side %}
-{% endblock %}
-
-{% block breadcrumbs %}
-{% endblock %}
+{% endblock %} {% block outer_side %} {% endblock %} {% block breadcrumbs %} {% endblock %}
```
Comment (johngian) [inline kitsune/landings/jinja2/landings/home.html]: Same here, lets split this to multiple lines.

File: kitsune/landings/jinja2/landings/includes/featured_articles.html [hunk 3]
```diff
@@ -1,93 +1,29 @@
...
+      <div class="card--details">
+        <h3 class="card--title">
+          <a
+            class="expand-this-link"
+            href="{{ url('wiki.document', doc.slug)  }}"
+            data-event-category="link click"
+            data-event-action="product"
+            data-event-label="#{{ doc.title }}"
+          >
+            {{ doc.title }}
```
Comment (johngian) [inline kitsune/landings/jinja2/landings/includes/featured_articles.html]: Indent pls

File: kitsune/wiki/utils.py [hunk 4]
```diff
@@ -116,10 +124,35 @@ def _active_contributors_id(from_date, to_date, locale, product):
...
+        featured = featured.filter(document__products__in=[product.id])
+
+    # Limit this to 10 entries
+    if featured.count() < 10:
+        return featured
+
+    # This returns a WikiDocumentVisits object and not the actual Wiki doc.
+    # The wiki doc it's under .document which is prefetched b/c of select_related
+    featured = featured[:10]
+    return random.sample(featured, 4)
```
Comment (johngian) [inline kitsune/wiki/utils.py]: nit: Here you can use `order_by('?')` to get the random order

Comment (akatsoulas) [reply kitsune/wiki/utils.py]: You cannot reorder the queryset after you slicing thus the workaround with random.

File: kitsune/landings/jinja2/landings/home.html (hunk 1 above)
Comment (akatsoulas) [reply kitsune/landings/jinja2/landings/home.html]: That's formatter's default behavior. I will tune it a bit

**Pull request merged**